src/datasource.ts

```diff
@@ -329,20 +329,96 @@ export default class GoogleStackdriverDatasource {
     });
   }
 
-  getMetricLabel(aliasPattern, series) {
-    let aliasRegex = /\{\{(.+?)\}\}/g;
+  filterSeries(target, response) {
+    if (!_.has(target, 'seriesFilter') ||
+        target.seriesFilter.mode === 'NONE' ||
+        target.seriesFilter.type === 'NONE' ||
+        target.seriesFilter.param === '') {
+      return response;
+    }
+
+    let param = _.toNumber(target.seriesFilter.param);
+    if (_.isNaN(param)) return response;
+
+    response.timeSeries.forEach(series => {
+      series['filterValue'] = this.getSeriesFilterValue(target, series);
+    });
+
+    switch(target.seriesFilter.mode) {
+    case "TOP":
+      response.timeSeries.sort(function(a, b) {
+        return b.filterValue - a.filterValue;
+      });
+      response.timeSeries = response.timeSeries.slice(0, param);
+      return response;
+    case "BOTTOM":
+      response.timeSeries.sort(function(a, b) {
+        return a.filterValue - b.filterValue;
+      });
+      response.timeSeries = response.timeSeries.slice(0, param);
+      return response;
+    case "BELOW":
+      response.timeSeries = response.timeSeries.filter(function(elem) {
+        return elem.filterValue < param;
+      });
+      return response;
+    case "ABOVE":
+      response.timeSeries = response.timeSeries.filter(function(elem) {
+        return elem.filterValue > param;
+      });
+      return response;
+    }
+  }
+
+  getSeriesFilterValue(target, series) {
+    let valueKey = series.valueType.toLowerCase() + 'Value';
+    switch(target.seriesFilter.type) {
+    case "MAX":
+      return series.points.reduce(function(acc, elem) {
+        return Math.max(acc, elem.value[valueKey]);
+      }, Number.MIN_VALUE);
+    case "MIN":
+      return series.points.reduce(function(acc, elem) {
+        return Math.min(acc, elem.value[valueKey]);
+      }, Number.MAX_VALUE);
```

Comment:
What should min/max do for an empty time series?

Reply:
That's a good question. I think that empty series should always be first to be filtered out, so for TOP and ABOVE mode it should be MIN_VALUE and for BOTTOM and BELOW it should be MAX_VALUE.